libsql-sqlite3/src/wal.c

```diff
@@ -1729,6 +1766,51 @@ static int walIteratorInit(Wal *pWal, u32 nBackfill, WalIterator **pp){
   return rc;
 }
 
+static int walIteratorRevInit(Wal *pWal, u32 nBackfill, struct WalIteratorRev *p, u32 mxSafeFrame, int getMissingFrame){
+    WalIterator *pIter;
+    u32 *frames;
+    u32 iFrame, iPageno;
+    int rc;
+
+    frames = (u32*)sqlite3MallocZero((pWal->hdr.mxFrame + 1) * sizeof(u32));
```

Comment:
note - this is correct because sqlite3MallocZero aligns to 8 bytes (as per https://www.sqlite.org/c3ref/free.html). I'm posting this comment here just to mark that I verified that